doc/migration.rst

```diff
@@ -114,3 +131,42 @@ it is not preserved for `dune-workspace` files. I.e. a
 `dune-workspace.in` file means nothing.
 
 This rule was only useful when we didn't have project files.
+
+Variable Syntax
+---------------
+
+``${foo} and $(foo)`` are no longer valid variable syntax in dune files.
+Variables are defined as ``%{foo}``. This change is done to simplify
+interoperability with bash commands which also use teh ``${foo}`` syntax.
+
+``(files_recursively_in ..)`` is removed
+----------------------------------------
+
+The ``files_recursively_in`` dependency specification is invalid in dune files.
+A :ref:`source_tree <source_tree>` stanza has been introduced to reflect the
+actual function of this stanza.
+
+Escape Sequences
+----------------
+
+Invalid escape sequences of the form ``\x`` where ``x`` is a character other
+than ``[0-9]``, ``x``, ``n``, ``r``, ``t``, ``b``.
```

Comment:
it looks like this sentence is not finished

Reply:
Oops, finished that.